handler/src/test/java/io/netty/handler/ssl/JdkConscryptSslEngineInteropTest.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.ssl;
+
+import java.security.Provider;
+import org.conscrypt.OpenSSLProvider;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static org.junit.Assume.assumeTrue;
+
+@RunWith(Parameterized.class)
+public class JdkConscryptSslEngineInteropTest extends SSLEngineTest {
+
+    @Parameterized.Parameters(name = "{index}: bufferType = {0}")
+    public static Collection<Object> data() {
+        List<Object> params = new ArrayList<Object>();
+        for (BufferType type: BufferType.values()) {
+            params.add(type);
+        }
+        return params;
+    }
+
+    public JdkConscryptSslEngineInteropTest(BufferType type) {
+        super(type);
+    }
+
+    @BeforeClass
+    public static void checkConscrypt() {
+        assumeTrue(ConscryptAlpnSslEngine.isAvailable());
+    }
+
+    @Override
+    protected SslProvider sslClientProvider() {
+        return SslProvider.JDK;
+    }
+
+    @Override
+    protected SslProvider sslServerProvider() {
+        return SslProvider.JDK;
+    }
+
+    @Override
+    protected Provider serverSslContextProvider() {
+        return new OpenSSLProvider();
+    }
+
+    @Override
+    @Test
+    public void testMutualAuthInvalidIntermediateCASucceedWithOptionalClientAuth() throws Exception {
+        super.testMutualAuthInvalidIntermediateCASucceedWithOptionalClientAuth();
+    }
+
+    @Override
+    @Test
+    public void testMutualAuthInvalidIntermediateCAFailWithOptionalClientAuth() throws Exception {
+        super.testMutualAuthInvalidIntermediateCAFailWithOptionalClientAuth();
+    }
+
+    @Override
+    @Test
+    public void testMutualAuthInvalidIntermediateCAFailWithRequiredClientAuth() throws Exception {
+        super.testMutualAuthInvalidIntermediateCAFailWithRequiredClientAuth();
+    }
+
+    @Override
+    @Test
+    @Ignore("TODO: Make this work with Conscrypt")
+    public void testMutualAuthValidClientCertChainTooLongFailOptionalClientAuth() throws Exception {
+        super.testMutualAuthValidClientCertChainTooLongFailOptionalClientAuth();
+    }
+
+    @Override
+    @Test
+    @Ignore("TODO: Make this work with Conscrypt")
+    public void testMutualAuthValidClientCertChainTooLongFailRequireClientAuth() throws Exception {
+        super.testMutualAuthValidClientCertChainTooLongFailRequireClientAuth();
+    }
+
+    @Override
+    protected boolean mySetupMutualAuthServerIsValidException(Throwable cause) {
```

Comment:
> IIRC I was seeing that the exception was wrapped in a DecoderException

I wouldn't expect a `DecoderException` because these tests directly use the `SSLEngine`. If you run `testMutualAuthInvalidIntermediateCAFailWithOptionalClientAuth` with this test you will see issue #6421. It looks like conscrypt behaves like the JDK and throws a SSLException instead of a SSLHandshakeException. Can you (or a conscrypt contributor) comment if this is intentional, or just to be the same as the jdk, and do you think #6421 is a valid issue?